File: iks-core/src/com/iks/cms/core/sql/ColumnValue.java
package com.iks.cms.core.sql;

import org.apache.commons.lang3.*;

/**
 * @author Igor Kaynov
 */
public class ColumnValue {
  private Object     value = null;
  private ColumnType type  = ColumnType.String;
  public ColumnValue( Object value ) {
    this.value = value;
  }
  public ColumnValue( Object value, ColumnType type ) {
    this( value );
    this.type = type;
  }
  public Object getValue() {
    return value;
  }
  public void setValue( Object value ) {
    this.value = value;
  }
  public String toString() {
    switch( type ) {
      case String:
        String valueStr = StringUtils.trimToNull( value.toString() );
        return valueStr == null ? null : "'" + valueStr + "'";
      default:
        return value.toString();
    }
  }
}

File: iks-core/src/com/iks/cms/core/sql/ColumnValue.kt
package com.iks.cms.core.sql

import org.apache.commons.lang3.*

/**
 * @author Igor Kaynov
 */
class ColumnValue(value: Any) {
  var value: Any? = null
  var type = ColumnType.String

  init {
    this.value = value
  }

  constructor(value: Any, type: ColumnType) : this(value) {
    this.type = type
  }

  override fun toString(): String {
    return value?.toString() ?: "null"
  }
}
